block/retriever.go

```diff
@@ -103,6 +103,12 @@ func (m *Manager) applyLocalBlock(height uint64) error {
 	}
 	m.retrieverMu.Unlock()
 
+	// validate configuration params and rollapp consensus params keep in line
+	err = m.ValidateConfigWithRollappParams()
```

Comment:
why not just have this logic as a first thing in the `applyBlock` vs putting it in 2 places?  (currently both in `applyLocalBlock` and `produceBlock`)